encode.go

```diff
@@ -441,6 +442,9 @@ func (en *encoder) sliceReflect(key uint64, slval reflect.Value) {
 		en.Write(b)
 		return
 
+	case reflect.Slice:
+		panic("protobuf: slice of slice is not supported")
```

Comment:
The panic is actually catched at the top most public method, which returns an error. Quick way to propagate an error, I just followed the path the library already uses.